src/wallet/wallet_rpc_server.cpp

```diff
@@ -2508,6 +2543,105 @@ namespace tools
     }
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::try_auto_renew()
+  {
+    std::string error;
+    const uint64_t staking_balance  = m_wallet->unlocked_balance(0);
+    const uint64_t required_balance = xx__get_staking_requirement();
+    const uint64_t curr_height      = m_wallet->get_daemon_blockchain_height(error);
+
+    if (!error.empty())
+    {
+      LOG_ERROR(tr("Error getting daemon blockchain height") + error);
+      return false;
+    }
+
+    if (staking_balance > required_balance)
```

Comment:
Probably a good idea to make this greater than a certain window around the required balance, so that if the requirement changes (e.g. somebody else stakes or if your staking tx makes it into a block a few blocks later than you expected), then this algorithm still works.
```
cant stake    stake        remove excess
-------------|-----|------------------
    required ^     ^ required+threshold
```
@tewinget this is a reason for checking both `locktime >= required locktime`, and `balance >= required balance`. I believe the required balance may only go down, so a tx taking longer to get into a block couldn't prevent you from registering in that way and `required staking amount` going down isn't a problem.

However, lock time is specified as an exact height, so it's possible that you make your staking TX and then it takes one block too many to get into the chain and then the staking TX is useless... that would really suck for a user who did all the right steps to stake. So I guess that the users' wallet should specify a lock time that it greater than the staking requirement + some threshold, so that even if there is a little bit of a delay getting into a block, then the staking tx is still valid.